### ISSUE TITLE
Dots Partially Hidden with parent element's overflow set to hidden

### ISSUE THREAD

**Issue by ericdrobinson** [NightCatSama/vue-slider-component#337]
#### Describe the bug

Dots get partially cut off when the slider's parent element has `overflow: hidden;` set.

#### Additional context

The current padding calculation appears to differentiate between horizontal and vertical slider modes, applying padding in only the cross-axis direction (presumably for "stacking"). However, this does not take into account the fact that the dots will extend _beyond the bounds of the container_ in the "lengthwise-axis" of the rail unless extra padding is applied.

[This JSFiddle](https://jsfiddle.net/0kf5wvn6/4/) shows the issue:
![image](https://user-images.githubusercontent.com/9142587/55421582-8aba1280-5547-11e9-8037-d6fc1cf75531.png)

In the above example, the first slider uses the current default padding settings. The second slider swaps the "horizontal" and "vertical" padding axes (which is obviously wrong). The third slider is set to have identical horizontal and vertical padding (as the dot is a circle).

I believe that [this](https://github.com/NightCatSama/vue-slider-component/blob/f9fe6e441031641ac942fd97da0a66aee2e96f3d/lib/vue-slider.tsx#L206) is the offending line of code. A simple fix would be to replace the above with the following logic:

```ts
padding: `${dotHeight / 2}px ${dotWidth / 2}px`,
```

#### Environment

 - Vue version: v2.6.10
 - Component Version: 3.0.22

**Comment by NightCatSama** [comment]
In fact, it used to be what you said.

Because if the component uses horizontal padding, it will cause the head and tail of the component to be out of alignment with other elements. [#319](<https://github.com/NightCatSama/vue-slider-component/issues/319>)

You can add `padding` to the container, for example: <https://jsfiddle.net/Lphwac7q/>

**Comment by ericdrobinson** [comment]
>In fact, it used to be what you said.

Yup. I noticed the issue when we updated to 3.x.

>Because if the component uses horizontal padding, it will cause the head and tail of the component to be out of alignment with other elements. #319

Sure. I wouldn't say that it's "out of alignment with other elements", but rather that the "dots" may not appear to be "centered" with respect to the edges of other elements. This is a design choice. In our case, it appears _wrong_.

>You can add `padding` to the container, for example: https://jsfiddle.net/Lphwac7q/

This requires adding a container element for the slider itself. Would it not be possible to update the padding logic to support _both_ scenarios? Perhaps by adding a `boolean contained` prop that specifies whether or not the slider should be _fully_ contained within its containing element?

Theoretically, support for this would look like:

```ts
padding: this.contained ? `${dotHeight / 2}px ${dotWidth / 2}px` : (this.isHorizontal ? `${dotHeight / 2}px 0` : `0 ${dotWidth / 2}px`),
```

This would allow for the following:

1. Explicit documentation of the way that the dots may extend outside of the bounds of the containing element (default).
1. A clean and simple way to have the slider draw correctly _without_ adding another element to the DOM hierarchy.
